### PR TITLE
ConfParser.py: Add permission error handler

### PR DESCRIPTION
**Pull request by madhukar01**
Add IOError handlers for code involving IO operations.

Fixes https://github.com/coala/coala/issues/4110

**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

### Checklist

- [ ] I read the [commit guidelines](http://coala.io/commit) and I've followed
      them.
- [ ] I ran coala over my code locally. (*All commits have to pass
      individually.* It is not sufficient to have "fixup commits" on your PR,
      our bot will still report the issues for the previous commit.) You will
      likely receive a lot of bot comments and build failures if coala does not
      pass on every single commit!

After you submit your pull request, **DO NOT click the 'Update Branch' button.**
When asked for a rebase, consult [coala.io/rebase](https://coala.io/rebase)
instead.

Please consider helping us by reviewing other peoples pull requests as well:

- pick up any PR at <https://coala.io/review>
- review it (check <https://coala.io/reviewing> for more info)
- if you are sure that it needs work, use `cobot mark wip <URL>` to get it out
  of the review queue.

The more you review, the more your score will grow at coala.io and we will
review your PRs faster!

### CODE REVIEW

File: coalib/parsing/ConfParser.py [hunk 1]
```diff
@@ -48,8 +48,11 @@ def parse(self, input_data, overwrite=False):
         if os.path.isdir(input_data):
             input_data = os.path.join(input_data, Constants.default_coafile)
 
-        with open(input_data, 'r', encoding='utf-8') as _file:
-            lines = _file.readlines()
+        try:
+            with open(input_data, 'r', encoding='utf-8') as _file:
+                lines = _file.readlines()
+        except EnvironmentError as e:
+            print("An error occurred: ", e)
```
Comment (adhikasp) [inline coalib/parsing/ConfParser.py]: Use `logging.error` not `print`

Comment (adhikasp) [inline coalib/parsing/ConfParser.py]: Also, have you tried to reproduce the error? If the configuration files is not loaded here but the error is already handled, does coala run normally?

Comment (madhukar01) [inline coalib/parsing/ConfParser.py]: Will do

**Comment by adhikasp** [comment]
@madhukar01 test are failing, please keep the coverage. Also could you confirm that coala still run fine after handling the error? What would happen to `.coarc` configuration gathering?

**Comment by madhukar01** [comment]
@adhikasp  Yeah, the change in this file is leading to errors in other tests. I'm working on fixing it :)

**Comment by nityeshaga** [comment]
@madhukar01 You still working on this?

**Comment by jayvdb** [comment]
@madhukar01 , you only need to add a test which causes an IOError .

This is one of our oldest open PRs.  Please finish it or we will need to unassign you from the task.

**Pull request closed without merging**